langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/common/VertexAiGeminiStreamingChatModelIT.java

```diff
@@ -0,0 +1,31 @@
+package dev.langchain4j.model.vertexai.common;
+
+import dev.langchain4j.model.chat.StreamingChatLanguageModel;
+import dev.langchain4j.model.chat.common.StreamingChatLanguageModelIT;
+import dev.langchain4j.model.vertexai.VertexAiGeminiStreamingChatModel;
+
+import java.util.List;
+
+class VertexAiGeminiStreamingChatModelIT extends StreamingChatLanguageModelIT {
+
+    @Override
+    protected List<StreamingChatLanguageModel> models() {
+        return List.of(
+                VertexAiGeminiStreamingChatModel.builder()
+                        .project(System.getenv("GCP_PROJECT_ID"))
+                        .location(System.getenv("GCP_LOCATION"))
+                        .modelName("gemini-1.5-flash")
+                        .build()
+        );
+    }
+
+    @Override
+    protected boolean assertFinishReason() {
+        return false; // TODO fix
+    }
+
+    @Override
+    protected boolean assertThreads() {
+        return false; // TODO fix
```

Comment:
Does it mean Vertex AI Gemini is using sometimes different threads when calling `onNext()`?
That said, as I mentioned earlier, I'm not sure why it should be mandatory to use just a single thread (even if that's a bit unexpected). For models that come with their own SDK, there's not much we can do if they use a pool of threads there.

Reply:
No, AFAIK, Vertex AI Gemini is the only one that uses the same thread (the one which called the model), because it processes tokens in a blocking manner. It does not mean it is bad, jsut that there is a discrepancy and user might be surprised. So we need to think about it again if it is OK or not. And if it is not OK, how to "fix" it.

Reply:
OK, I see, interesting...
My personal opinion is that I don't think we should care about the thread being used though.

Reply:
Our users do care, so I am afraid we should care as well. Our API contract should explicitly specify threading details.